### PR TITLE
Add python diagnostic

### PR DESCRIPTION
**Pull request by sfc-gh-cconner**
### Pre-review checklist
   * [* ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ *] I've added or updated automated unit tests to verify correctness of my new code.
   * [ *] I've added or updated integration tests to verify correctness of my new code.
   * [* ] I've confirmed that my changes are working by executing CLI's commands manually.
   * [* ] I've confirmed that my changes are up-to-date with the target branch.
   * [ * ] I've described my changes in the release notes.
   * [ * ] I've described my changes in the section below.

### Changes description
Added the ability to use these python connector options in the "snow connection test" command:

enable_connection_diag
connection_diag_log_path
connection_diag_whitelist_path

Now if you run:

snow connection test --enable-diag

You will get a connectivity report in the output table.

### CODE REVIEW

File: src/snowflake/cli/plugins/connection/commands.py [hunk 1]
```diff
@@ -213,13 +215,64 @@ def add(
...
+                enable_connection_diag="true",
+                connection_diag_log_path=diag_log_path,
+                connection_diag_whitelist_path=diag_allowlist_path,
+            )
+        else:
+            conn = connect_to_snowflake(
+                connection_name=connection,
+                enable_connection_diag="true",
+                connection_diag_log_path=diag_log_path,
+            )
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/connection/commands.py:272]: Let's encapsulate this into a method

File: src/snowflake/cli/plugins/connection/commands.py [hunk 2]
```diff
@@ -213,13 +215,64 @@ def add(
...
-def test(connection: str = ConnectionOption, **options) -> CommandResult:
+def test(
+    connection: str = ConnectionOption,
+    enable_diag: bool = typer.Option(
+        False,
+        "--enable-diag",
+        help="Run python connector diagnostic test.",
+        show_default=False,
+    ),
+    diag_log_path: str = typer.Option(
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/connection/commands.py:226]:
```suggestion
    diag_log_path: Path = typer.Option(
```
We can also use `typer.Option` arguments `file_ok` and `dir_ok` to do early evaluation if the path is valid one

File: src/snowflake/cli/plugins/connection/commands.py (hunk 1 above)
Comment (sfc-gh-cconner) [reply src/snowflake/cli/plugins/connection/commands.py]: Done

File: src/snowflake/cli/plugins/connection/commands.py (hunk 2 above)
Comment (sfc-gh-cconner) [reply src/snowflake/cli/plugins/connection/commands.py]: Done

**Comment by sfc-gh-turbaszek** [comment]
Closing as duplicate of #893

**Pull request closed without merging**